bigcache.go

```diff
@@ -33,6 +33,19 @@ type cacheShard struct {
 	onRemove    func(wrappedEntry []byte)
 }
 
+// EntryInfo holds informations about entry in the cache
+type EntryInfo struct {
+	Key       string
+	Hash      uint64
+	Timestamp uint64
+}
+
+// EntryInfoFilter function allows filtering of EntryInfo elements in the KeysIterator functions
+type EntryInfoFilter func(EntryInfo) bool
+
+// EntryInfoIterator allows to iterate over entries in the cache
+type EntryInfoIterator chan EntryInfo
```

Comment:
With channel you add blocking. Take a look at loops you created. You are adding mutex for reading from map and then pushing value to channel which is blocking. If you remove channel and instead return closure to function that returns entry form shard `i` on position `index` then you don't need to copy all (1024) elements and this will be a lazy iterator.